File: eco-api/src/main/java/com/willfp/eco/util/EntityUtils.java
package com.willfp.eco.util;

import com.willfp.eco.core.Eco;
import net.kyori.adventure.text.Component;
import org.bukkit.entity.LivingEntity;
import org.bukkit.entity.Player;
import org.jetbrains.annotations.NotNull;

/**
 * Utilities / API methods for entities.
 */
public final class EntityUtils {
    /**
     * Set a client-side entity display name.
     *
     * @param entity  The entity.
     * @param player  The player.
     * @param name    The display name.
     * @param visible If the display name should be forcibly visible.
     */
    public static void setClientsideDisplayName(@NotNull final LivingEntity entity,
                                                @NotNull final Player player,
                                                @NotNull final Component name,
                                                final boolean visible) {
        Eco.get().setClientsideDisplayName(entity, player, name, visible);
    }

    private EntityUtils() {
        throw new UnsupportedOperationException("This is a utility class and cannot be instantiated");
    }
}

File: eco-api/src/main/kotlin/com/willfp/eco/util/EntityUtils.kt
@file:JvmName("EntityUtilsExtensions")

package com.willfp.eco.util

import net.kyori.adventure.text.Component
import org.bukkit.entity.LivingEntity
import org.bukkit.entity.Player

/**
 * Set the display name of this entity as seen by a single player.
 *
 * The change is client-side only: it is sent to the given player and does not modify the
 * entity's actual display name on the server or for anyone else.
 *
 * @param player      The player to send the display name to.
 * @param displayName The display name to show.
 * @param visible     If the display name should be forcibly visible.
 * @see EntityUtils.setClientsideDisplayName
 */
fun LivingEntity.setClientsideDisplayName(player: Player, displayName: Component, visible: Boolean) {
    EntityUtils.setClientsideDisplayName(this, player, displayName, visible)
}
